### PR TITLE
change `raco pkg new` to generate both MIT and APACHE LICENSE files (replacing LGPL)

### PR DESCRIPTION
**Pull request by spdegabrielle**

### CODE REVIEW

File: racket/collects/pkg/private/new.rkt [hunk 1]
```diff
@@ -62,8 +62,8 @@
     (make-directory name)
     (parameterize ([current-directory name])
 
-      ;; LICENSE.txt
-      (with-output-to-file "LICENSE.txt"
+      ;; LICENSE files
+      (with-output-to-file "LICENSE-MIT"
```
Comment (sorawee) [inline racket/collects/pkg/private/new.rkt]: The text below is still that of LGPL

File: racket/collects/pkg/private/new.rkt [hunk 2]
```diff
@@ -77,6 +77,215 @@ particular means that you must release the source code for the
...
+      License. However, in accepting such obligations, You may act only
+      on Your own behalf and on Your sole responsibility, not on behalf
+      of any other Contributor, and only if You agree to indemnify,
+      defend, and hold each Contributor harmless for any liability
+      incurred by, or claims asserted against, such Contributor by reason
+      of your accepting any such warranty or additional liability.
+
+   END OF TERMS AND CONDITIONS
+
+   APPENDIX: How to apply the Apache License to your work.
```
Comment (sorawee) [inline racket/collects/pkg/private/new.rkt]: Please take the instruction appendix off (until "identification within third-party archives.")

Comment (spdegabrielle) [reply racket/collects/pkg/private/new.rkt]: done

File: racket/collects/pkg/private/new.rkt (hunk 1 above)
Comment (spdegabrielle) [reply racket/collects/pkg/private/new.rkt]: My mistake. Fixed.

**Comment by spdegabrielle** [comment]
@sorawee Thank you for reviewing so quickly.

File: racket/collects/pkg/private/new.rkt [hunk 3]
```diff
@@ -62,20 +62,229 @@
...
 
-      ;; LICENSE.txt
-      (with-output-to-file "LICENSE.txt"
+      ;; LICENSE files
+      (displayln "Generating LICENSE-APACHE and LICENSE-MIT files. You are free to change the license.")
+      (with-output-to-file "LICENSE-MIT"
+        (lambda () (expand/display #<<EOS
+MIT License
+
+Copyright (c) <<year>> <<name>> <<user>>
```
Comment (sorawee) [inline racket/collects/pkg/private/new.rkt]: I'm not sure what's the best way to deal with this line, but the lack of punctuation between `<<name>>` and `<<user>>` looks... incorrect.

```
Copyright (c) 2019 hello-world sorawee
```

Comment (spdegabrielle) [reply racket/collects/pkg/private/new.rkt]: fixed - now has name on first line

**Comment by spdegabrielle** [comment]
@samth licence file generation change suggested by @sorawee 
all mistakes are mine, thank you to @sorawee for my mistakes. And @greghendershott too.

Added  `(displayln "Generating LICENSE-APACHE and LICENSE-MIT files. You are free to change the license.")`

In line with the guidance at https://www.apache.org/licenses/LICENSE-2.0#apply I've changed the LICENSE-APACHE file generation.

File: racket/collects/pkg/private/new.rkt [hunk 4]
```diff
@@ -62,20 +62,51 @@
     (make-directory name)
     (parameterize ([current-directory name])
 
-      ;; LICENSE.txt
-      (with-output-to-file "LICENSE.txt"
+      ;; LICENSE files
+      (displayln "Generating LICENSE-APACHE and LICENSE-MIT files. You are free to change the license.")
```
Comment (bennn) [inline racket/collects/pkg/private/new.rkt]: Is there any reason these don't use the `.txt` file extension?

Comment (spdegabrielle) [reply racket/collects/pkg/private/new.rkt]: Hi @benn

@sorawee asked @samth and Sam said no to having a txt extension.
 
See https://github.com/racket/racket/issues/1570#issuecomment-501449180

Kind regards, 
Stephen

**Comment by spdegabrielle** [comment]
Updated the generated main.rkt file to refer to the new license files included.

File: racket/collects/pkg/private/new.rkt [hunk 5]
```diff
@@ -222,9 +222,8 @@ EOS
 ;; To view documentation:
 ;;   $ raco docs <<name>>
 ;;
-;; For your convenience, we have included a LICENSE.txt file, which links to
-;; the GNU Lesser General Public License.
```
Comment (NlightNFotis) [inline racket/collects/pkg/private/new.rkt]: Is this bit of information user facing somewhere? Perhaps `raco` output? or is it only a source code comment?

Comment (NlightNFotis) [reply racket/collects/pkg/private/new.rkt]: Ignore this comment, I just noticed in an earlier commit that this is user facing. It's weird though that the comment is down below. Is there any reason for it being located where it is, or is it misplaced?

Comment (spdegabrielle) [reply racket/collects/pkg/private/new.rkt]: for other reviewers - this section was also changed to refer to MIT and APACHE licenses,

``` 
;; To view documentation:
;;   $ raco docs <<name>>
;;
;; For your convenience, we have included LICENSE-MIT and LICENSE-APACHE files.
;; If you would prefer to use a different license, replace those files with the
;; desired license.
```

**Comment by NlightNFotis** [comment]
Also, before this gets merged in, can the multiple `Update new.rkt` commits be squashed into one with a more descriptive commit message?

**Comment by spdegabrielle** [comment]
squashed into new PR #2701

**Pull request closed without merging**